lib/Serialization/ModuleFile.h

```diff
@@ -146,7 +146,7 @@ class ModuleFile
     }
 
     bool isLoaded() const {
-      return Import.second != nullptr;
+      return Import.hasValue() && Import.getValue().importedModule != nullptr;
```

Comment:
Bit subjective but... both here and in other places where we are projecting immediately after doing `getValue()`, you use the `operator->` overload to combine those.

So `return Import.hasValue() && Import.getValue().importedModule != nullptr;` would become

```
return Import.hasValue() && Import->importedModule != nullptr;
```

Some people on the team might even use the implicit conversion to `bool` and write it as 

```
return Import && Import->importedModule != nullptr;
```

but that seems like a bit too much magic because one might guess that `Import` is a pointer even though it's not a pointer.

Reply:
Resolved in 1d625d8, I think. There is one place where I did *(foo.Import) to extract the value after an assertion check that it had a value. That looks like a pointer but is probably OK.